test/test_sparse_datastore.py

```diff
@@ -1,10 +1,12 @@
 """Test framers."""
 
+import pytest
 
 from pymodbus.datastore import ModbusSparseDataBlock
 
 
-def test_check_sparsedatastore():
+@pytest.mark.asyncio()
```

Comment:
Please leve the old methods.

Instead please add an extra test with async (e.g. duplicate the existing). Since this PR adds async but do not remove sync, it is important to test that both variants work.

Reply:
sure, I can duplicate the test function and make it separate